### DISCUSSION TITLE
Allow setting axis colors

### DISCUSSION THREAD

**Discussion by bduffany** [grafana/grafana#38193] [Dashboards, Panels & Visualizations]
**What would you like to be added**: ability to set label & number colors of axes

**Why is this needed**: When you have two series and two axes on the same graph, it is difficult to tell which series corresponds to which axis.

Currently, the cognitive process is like this:
* Look at a series; it's green
* What is the current value of the green series? Hmm, not sure which axis to look at.
* Let me check the legend -- ah, I see that the green series corresponds to "foo."
* Let me find the axis which is labeled "foo."
* Ah, I see that the value of "foo," the green series, is currently "5".

If axis colors were introduced, the cognitive process would be more like this:
* Look at a series, it's green
* What is the current value of the green series? Ah, must be the green axis. I see that the value is "5", and the label is "foo".

![Colored axis](https://user-images.githubusercontent.com/40671217/114753497-ceab9900-9d57-11eb-9836-e00921ee4392.jpg)

**Comment by leeoniya** [comment]
this will be supported by the underlying chart library in the new graph panel. e.g. https://leeoniya.github.io/uPlot/demos/axis-control.html

i think it's a valid concern, but i'm not sure if there are plans to expose this currently.

**Comment by dprokop** [comment]
@bduffany thanks for your contribution. I'm gonna mark this as a feature request. We currently do not plan to support such feature, but if there is more interest in it we will definitely take this into consideration.

**Comment by bduffany** [comment]
@leeoniya ooooo, migrating to a canvas-based plot I see! Super exciting! Should dramatically improve performance.

**Comment by leeoniya** [comment]
the current panel uses Flot, which is also canvas-based and, to be honest, has held up surprisingly well for its age. but yes the new chart will be much faster for cases like high frequency streaming and huge datasets.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by positron96** [comment]
Hi, has there been any progress by chance? I believe any chart with dual axis would benefit from this!

**Comment by leeoniya** [comment]
done in https://github.com/grafana/grafana/pull/51437
